From 813dfbd6845e5366f2dbef190a695f8b06c1d83d Mon Sep 17 00:00:00 2001
From: Takashi Sakamoto <o-takashi@sakamocchi.jp>
Date: Fri, 18 Oct 2019 15:19:09 +0900
Subject: [PATCH] ALSA: firewire-lib: cancel flushing isoc context in the laste step to process context callback
Git-commit: 813dfbd6845e5366f2dbef190a695f8b06c1d83d
Patch-mainline: v5.5-rc1
References: jsc#SLE-16518

The aim of AMDTP domain is to process several isoc context in the same
time. However, current implementation is against this idea because it
flushes each isoc context in the end of processing context callback.

This commit cancels it.

Signed-off-by: Takashi Sakamoto <o-takashi@sakamocchi.jp>
Link: https://lore.kernel.org/r/20191018061911.24909-5-o-takashi@sakamocchi.jp
Signed-off-by: Takashi Iwai <tiwai@suse.de>

---
 sound/firewire/amdtp-stream.c | 4 ----
 1 file changed, 4 deletions(-)

diff --git a/sound/firewire/amdtp-stream.c b/sound/firewire/amdtp-stream.c
index 3d27d4ce2b45..36c3f1f9dbff 100644
--- a/sound/firewire/amdtp-stream.c
+++ b/sound/firewire/amdtp-stream.c
@@ -842,8 +842,6 @@ static void out_stream_callback(struct fw_iso_context *context, u32 tstamp,
 	}
 
 	s->event_count = event_count;
-
-	fw_iso_context_queue_flush(s->context);
 }
 
 static void in_stream_callback(struct fw_iso_context *context, u32 tstamp,
@@ -897,8 +895,6 @@ static void in_stream_callback(struct fw_iso_context *context, u32 tstamp,
 	}
 
 	s->event_count = event_count;
-
-	fw_iso_context_queue_flush(s->context);
 }
 
 /* this is executed one time */
-- 
2.16.4

